codec/src/main/java/io/netty5/handler/codec/Delimiters.java

```diff
@@ -27,19 +28,19 @@
      * Returns a {@code NUL (0x00)} delimiter, which could be used for
      * Flash XML socket or any similar protocols.
      */
-    public static ByteBuf[] nulDelimiter() {
-        return new ByteBuf[] {
-                Unpooled.wrappedBuffer(new byte[] { 0 }) };
+    public static Buffer[] nulDelimiter() {
+        return new Buffer[] {
+                onHeapUnpooled().copyOf(new byte[] { 0 }).makeReadOnly() };
     }
 
     /**
      * Returns {@code CR ('\r')} and {@code LF ('\n')} delimiters, which could
      * be used for text-based line protocols.
      */
-    public static ByteBuf[] lineDelimiter() {
-        return new ByteBuf[] {
-                Unpooled.wrappedBuffer(new byte[] { '\r', '\n' }),
-                Unpooled.wrappedBuffer(new byte[] { '\n' }),
+    public static Buffer[] lineDelimiter() {
+        return new Buffer[] {
+                onHeapUnpooled().copyOf(new byte[] { '\r', '\n' }).makeReadOnly(),
+                onHeapUnpooled().copyOf(new byte[] { '\n' }).makeReadOnly(),
```

Comment:
I noticed that in the places where we use these methods, we never close these buffers.

Reply:
let me fix this

Reply:
Sorry to bump this older conversation, but wouldn't it also be better to use `DefaultBufferAllocators` here? If I understand this correctly then the buffers are closed in the decoders, but the allocators aren't.